proposals/0004-open-api.md

```
@@ -0,0 +1,446 @@
+| Issue     | <https://youtrack.jetbrains.com/issue/KTOR-8316/>                                                  |
+| Prototype | <https://github.com/ktorio/ktor-build-plugins/tree/bjhham/openapi-eap/samples/ktor-openapi-sample> |
+
+### Contents
+
+1. [Summary](#summary)
+2. [Motivation](#motivation)
+3. [Current Solutions](#current-solutions)
+4. [Design Overview](#design-overview)
+5. [Design Details](#design-details)
+6. [Technical Details](#technical-details)
+    1. [Routing API Introspection](#routing-api-introspection)
```

markdownlint-cli2 (0.17.2)

Link fragments should be valid

(MD051, link-fragments)